subscriber_notifee.go

```diff
@@ -14,82 +16,201 @@ import (
 // subscriberNotifee implements network.Notifee and also manages the subscriber to the event bus. We consume peer
 // identification events to trigger inclusion in the routing table, and we consume Disconnected events to eject peers
 // from it.
-type subscriberNotifee IpfsDHT
-
-func (nn *subscriberNotifee) DHT() *IpfsDHT {
-	return (*IpfsDHT)(nn)
+type subscriberNotifee struct {
+	dht                                    *IpfsDHT
+	identifySub, updateSub, routabilitySub event.Subscription
```

Comment:
We can also just have a single subscription for all of these then switch on the type. Actually, we should definitely do that for the identify/update subs. Otherwise, we could get the events out of order (identify after protocol update).

Reply:
Note: to subscribe to multiple events, call `Subscribe([]interface{}{event1, event2, etc...})`.

Reply:
SGTM